Extending the DCGAN example implemented by gluon API to provide a more straight-forward evaluation on the generated image (#12790)
apache/incubator-mxnet: example/gluon/DCGAN/README.md

# DCGAN in MXNet

[Deep Convolutional Generative Adversarial Networks(DCGAN)](https://arxiv.org/abs/1511.06434) implementation with Apache MXNet GLUON.
This implementation uses [inception_score](https://github.com/openai/improved-gan) to evaluate the model.

You can use this reference implementation on MNIST and CIFAR10 dataset.

#### Generated pic(use dataset cifar10)
![Generated pic](https://github.com/pengxin99/incubator-mxnet/blob/dcgan-inception_score/example/gluon/DCGAN/pic/fake_img_iter_13900.png)

> #### Generated pic(use dataset mnist)

Reviewer: ```suggestion
#### Generated image output examples from the MNIST dataset
```

> ![Generated image output examples from the CIFAR-10 dataset](https://github.com/dmlc/web-data/tree/master/mxnet/example/gluon/DCGAN/fake_img_iter_13900.png)

Reviewer: ```suggestion
![Generated image output examples from the CIFAR-10 dataset](https://raw.githubusercontent.com/dmlc/web-data/master/mxnet/example/gluon/DCGAN/fake_img_iter_13900.png)
```

> ![Generated pic](https://github.com/pengxin99/incubator-mxnet/blob/dcgan-inception_score/example/gluon/DCGAN/pic/fake_img_iter_21700.png)

Reviewer: ```suggestion
![Generated image output examples from the MNIST dataset](https://github.com/pengxin99/incubator-mxnet/blob/dcgan-inception_score/example/gluon/DCGAN/pic/fake_img_iter_21700.png)
```

> ![Generated image output examples from the MNIST dataset](https://github.com/dmlc/web-data/tree/master/mxnet/example/gluon/DCGAN/fake_img_iter_21700.png)

Reviewer: ```suggestion
![Generated image output examples from the MNIST dataset](https://raw.githubusercontent.com/dmlc/web-data/master/mxnet/example/gluon/DCGAN/fake_img_iter_21700.png)
```

#### inception_score in cpu and gpu (the real image`s score is around 3.3)
CPU & GPU

> ![inception_socre_with_cpu](https://github.com/pengxin99/incubator-mxnet/blob/dcgan-inception_score/example/gluon/DCGAN/pic/inception_score_cifar10_cpu.png)

Reviewer: ```suggestion
![inception score with CPU](https://github.com/pengxin99/incubator-mxnet/blob/dcgan-inception_score/example/gluon/DCGAN/pic/inception_score_cifar10_cpu.png)
```
When you put in the alt text like this, it becomes readable by screen readers and the like... and it's searchable.

> ![inception score with CPU](https://github.com/dmlc/web-data/tree/master/mxnet/example/gluon/DCGAN/inception_score_cifar10_cpu.png)

Reviewer: ```suggestion
![inception score with CPU](https://raw.githubusercontent.com/dmlc/web-data/master/mxnet/example/gluon/DCGAN/inception_score_cifar10_cpu.png)
```

> ![inception_score_with_gpu](https://github.com/pengxin99/incubator-mxnet/blob/dcgan-inception_score/example/gluon/DCGAN/pic/inception_score_cifar10.png)

Reviewer: ```suggestion
![inception score with GPU](https://github.com/pengxin99/incubator-mxnet/blob/dcgan-inception_score/example/gluon/DCGAN/pic/inception_score_cifar10.png)
```

> ![inception score with GPU](https://github.com/dmlc/web-data/tree/master/mxnet/example/gluon/DCGAN/inception_score_cifar10.png)

Reviewer: ```suggestion
![inception score with GPU](https://raw.githubusercontent.com/dmlc/web-data/master/mxnet/example/gluon/DCGAN/inception_score_cifar10.png)
```

## Quick start

> use below code to see the configurations you can set:

Reviewer: ```suggestion
Use the following code to see the configurations you can set:
```

> ```python

Reviewer: ```suggestion
```bash
```

      -h, --help            show this help message and exit
      --dataset DATASET     dataset to use. options are cifar10 and mnist.
      --batch-size BATCH_SIZE  input batch size, default is 64
      --nz NZ               size of the latent z vector, default is 100
      --ngf NGF             the channel of each generator filter layer, default is 64.
      --ndf NDF             the channel of each descriminator filter layer, default is 64.
      --nepoch NEPOCH       number of epochs to train for, default is 25.
      --niter NITER         save generated images and inception_score per niter iters, default is 100.
      --lr LR               learning rate, default=0.0002
      --beta1 BETA1         beta1 for adam. default=0.5
      --cuda                enables cuda
      --netG NETG           path to netG (to continue training)
      --netD NETD           path to netD (to continue training)
      --outf OUTF           folder to output images and model checkpoints
      --check-point CHECK_POINT
                            save results at each epoch or not
      --inception_score INCEPTION_SCORE
                            To record the inception_score, default is True.

> use below code to train DCGAN model with default configurations and dataset(cifar10), and metric with inception_score:

Reviewer: ```suggestion
Use the following Python script to train a DCGAN model with default configurations using the CIFAR-10 dataset and record metrics with `inception_score`:
```

> ```python

Reviewer: ```suggestion
```bash
```
